lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -23,6 +23,10 @@
  * You can not rely on the internal iterator of the collection being at a certain
  * position unless you explicitly positioned it before. Prefer iteration with
  * external iterators.
+ *
+ * @template T
+ * @template-extends IteratorAggregate<int|string, T>
+ * @template-extends ArrayAccess<int|string, T>
```

Comment:
```suggestion
 * @template-extends ArrayAccess<int|string|null, T>
```
Adding null to the list of key types will allow to push elements into colleciton with empty bracket syntax:
```php
$coll[] = $elt;
```